models/applicantModel.js

```diff
@@ -1,28 +1,461 @@
+    let conn;
+    try {
+      conn = await pool.getConnection();
+      await conn.beginTransaction();
+      console.log("Editing travel request with ID:", requestId);
+
+      // Destructure travel details from request body
+      const {
+        router_index,
+        notes,
+        requested_fee = 0,
+        imposed_fee = 0,
+        origin_country_name,
+        origin_city_name,
+        destination_country_name,
+        destination_city_name,
+        beginning_date,
+        beginning_time,
+        ending_date,
+        ending_time,
+        plane_needed,
+        hotel_needed,
+        additionalRoutes = [],
+      } = travelChanges;
+
+      // Format the routes into a single array
+      const allRoutes = formatRoutes(
+        {
+          router_index,
+          origin_country_name,
+          origin_city_name,
+          destination_country_name,
+          destination_city_name,
+          beginning_date,
+          beginning_time,
+          ending_date,
+          ending_time,
+          plane_needed,
+          hotel_needed,
+        },
+        additionalRoutes
+      );
+
+      // =======================================
+      // Step 1: Update Request table
+      // =======================================
+      const request_days = getRequestDays(allRoutes);
+
+      // Log old data
+      const [oldData] = await conn.query(
+        `SELECT * FROM request WHERE request_id = ?`,
+        [requestId]
+      );
+      console.log("Old data:", oldData);
+
+      const updateRequestTable = `
+        UPDATE request SET
+          notes = ?,
+          requested_fee = ?,
+          imposed_fee = ?,
+          request_days = ?,
+          last_mod_date = CURRENT_TIMESTAMP
+        WHERE request_id = ?
+      `;
+
+      await conn.execute(updateRequestTable, [
+        notes,
+        requested_fee, // Allow null values for requested_fee
+        imposed_fee, // Allow null values for imposed_fee
+        request_days, // Allow null values for request_days
+        requestId, // Use the provided requestId to update the correct record
+      ]);
+
+      // Log new data
+      const [newData] = await conn.query(
+        `SELECT * FROM request WHERE request_id = ?`,
+        [requestId]
+      );
+      console.log("New data:", newData);
+
+      // =======================================
+      // Step 2: Delete old routes
+      // =======================================
+
+      const oldRoutesIds = await conn.query(
+        `SELECT route_id FROM route_request WHERE request_id = ?`,
+        [requestId]
+      );
+
+      // Delete old route request table data related to the request
+      const deleteRouteRequest = `
+        DELETE FROM route_request WHERE request_id = ?
+      `;
+      await conn.execute(deleteRouteRequest, [requestId]);
+
+      // Delete old routes from Route_Request table
+      for (const route_id of oldRoutesIds) {
+        const deleteRoute = `
+          DELETE FROM route WHERE route_id = ?
```

Comment:
```suggestion
          DELETE FROM Route WHERE route_id = ?
```
Incorrect name for the table.